### PR TITLE
Binomial and Poisson Moment

### PR DESCRIPTION
**Pull request by farhanreynaldo**
Add moments and tests for the below distributions as part of https://github.com/pymc-devs/pymc/issues/5078:

* pymc.distributions.continuous.Binomial
* pymc.distributions.Discrete.Poisson

### CODE REVIEW

**Comment by twiecki** [comment]
@farhanreynaldo Need to resolve conflicts.

**Comment by farhanreynaldo** [comment]
> @farhanreynaldo Need to resolve conflicts.

This is my first time having a conflict. Are the following steps correct?

```
git fetch upstream
git rebase upstream/main
```

and then accept both changes and push the resolved conflicts?
```
git commit -m "resolve conflicts"
git push origin <branch-name>
```

**Comment by twiecki** [comment]
I think during rebasing you just git add, and then you force push.

File: pymc/distributions/discrete.py [hunk 1]
```diff
@@ -570,6 +576,11 @@ def dist(cls, mu, *args, **kwargs):
         # mode = intX(at.floor(mu))
```
Comment (ricardoV94) [inline pymc/distributions/discrete.py]: Seems like we have a leftover from before. Do you mind removing the commented out code?
```suggestion
```

Comment (ricardoV94) [reply pymc/distributions/discrete.py]: Also note that we used the floor of mu, we should do the same in the new `get_moment`

File: pymc/distributions/discrete.py [hunk 2]
```diff
@@ -117,6 +117,12 @@ def dist(cls, n, p, *args, **kwargs):
         # mode = at.cast(tround(n * p), self.dtype)
```
Comment (ricardoV94) [inline pymc/distributions/discrete.py]:
```suggestion
```

Same here, we should remove the old commented out code, and use `at.round` in the new `get_moment`

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Left a couple of suggestions @farhanreynaldo. Thanks for the PR!

File: pymc/distributions/discrete.py (hunk 1 above)
Comment (farhanreynaldo) [reply pymc/distributions/discrete.py]: Is the `IntX` necessary or should I just include the `at.floor`?

File: pymc/distributions/discrete.py (hunk 2 above)
Comment (farhanreynaldo) [reply pymc/distributions/discrete.py]: Is the `at.cast` also necessary here?

Comment (ricardoV94) [reply pymc/distributions/discrete.py]: Not anymore. That's done automatically by the function that calls get_moment

File: pymc/distributions/discrete.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/distributions/discrete.py]: Not needed either I think

File: pymc/tests/test_distributions_moments.py [hunk 3]
```diff
@@ -318,11 +326,44 @@ def test_gamma_moment(alpha, beta, size, expected):
...
+@pytest.mark.parametrize(
+    "mu, size, expected",
+    [
+        (2, None, 2),
+        (2, 5, np.full(5, 2)),
+        (np.arange(1, 5), None, np.arange(1, 5)),
+        (np.arange(1, 5), (2, 4), np.full((2, 4), np.arange(1, 5))),
+    ],
+)
+def test_poisson_moment(mu, size, expected):
```
Comment (ricardoV94) [inline pymc/tests/test_distributions_moments.py:366]: Some of the test combinations should include non integer values for mu to check the flooring is working as expected

File: pymc/tests/test_distributions_moments.py [hunk 4]
```diff
@@ -318,11 +326,44 @@ def test_gamma_moment(alpha, beta, size, expected):
...
+@pytest.mark.parametrize(
+    "n, p, size, expected",
+    [
+        (10, 0.5, None, 5),
+        (10, 0.5, 5, np.full(5, 5)),
+        (10, np.arange(1, 6) / 10, None, np.arange(1, 6)),
+        (10, np.arange(1, 6) / 10, (2, 5), np.full((2, 5), np.arange(1, 6))),
+    ],
+)
+def test_binomial_moment(n, p, size, expected):
```
Comment (ricardoV94) [inline pymc/tests/test_distributions_moments.py:351]: Some of the test combinations should have a `n*p` that is different when rounded and not rounded to make sure it is working as expected

**Review by ricardoV94 — COMMENTED** [review]
Some suggestions to make the tests more robust

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**